static-build/cmake/AddTarantoolProject.cmake

```diff
@@ -16,6 +16,8 @@ ExternalProject_Add(tarantool
         -DCMAKE_INSTALL_LOCALSTATEDIR=<INSTALL_DIR>/var
         -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
         -DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
+        # Ensure that no external dependencies are dynamically linked.
+        -DCMAKE_FIND_USE_CMAKE_SYSTEM_PATH=FALSE
```

Comment:
AFAIR, @CuriousGeorgiy had some specific reason to eliminate the option: something goes wrong with it. Sadly, I don't remember the exact reason. It worth to re-check it I guess.

Reply:
Just try to run static build on macOS, and the problem should probably manifestate itself.

Reply:
As far as I vaguely remember, we have a bug in the static build setup in that we link against the `.dylib` instead of the `.tlb` and this doesn’t work for system libraries like libunwind.

This didn’t manifestate itself before I refactored the libunwind library setup, but when it did, I just used this workaround for some reason.

Reply:
https://discourse.cmake.org/t/newer-versions-of-macos-require-linking-against-tbd-files-and-not-old-system-paths-to-dylib/6871

Reply:
But we link libunwind statically in the static build, correct? How .dylib/.tbd are involved?

Reply:
> But we link libunwind statically in the static build

libunwind on macOS is a system library (taken from the [LLVM project](https://github.com/llvm/llvm-project/tree/main/libunwind)), it has nothing to do with the [libunwind library](https://github.com/libunwind/libunwind) we use on Linux/FreeBSD.

Reply:
As far as I see right now from the comment above (can't check this right now), the problem is that we need to link against the `System` umbrella framework instead of the `libunwind` library, and we can't find the `System` umbrella framework if we set `CMAKE_FIND_USE_CMAKE_SYSTEM_PATH=FALSE`.

Reply:
Yes, reproduced on `osx_static_cmake`:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
LIBUNWIND_INCLUDE_DIR (ADVANCED)
```
I will close this PR.